selene/src/validate_config.rs

```diff
@@ -0,0 +1,123 @@
+use std::{
+    collections::HashMap,
+    path::{Path, PathBuf},
+};
+
+use selene_lib::CheckerConfig;
+use serde::Serialize;
+
+use crate::standard_library::StandardLibraryError;
+
+#[derive(Debug, Serialize)]
+pub struct InvalidConfigError {
+    error: String,
+    source: PathBuf,
+    range: Option<ErrorRange>,
+}
+
+#[derive(Debug, Serialize)]
+pub struct ErrorRange {
+    start: usize,
+    end: usize,
+}
+
+impl InvalidConfigError {
+    pub fn rich_output(self) -> String {
+        todo!("rich output for InvalidConfigError")
+    }
+}
+
+impl From<std::ops::Range<usize>> for ErrorRange {
+    fn from(range: std::ops::Range<usize>) -> Self {
+        Self {
+            start: range.start,
+            end: range.end,
+        }
+    }
+}
+
+impl From<serde_yaml::Location> for ErrorRange {
+    fn from(location: serde_yaml::Location) -> Self {
+        Self {
+            start: location.index(),
+            end: location.index(),
+        }
+    }
+}
+
+// TODO: Test
```

Comment:
Why not add the test in the same PR?

Reply:
All the TODOs will be solved before I undraft the PR, but I knew I'd forget if I didn't write it down